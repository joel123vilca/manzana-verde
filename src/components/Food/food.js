import { mapGetters, mapActions } from "vuex";

export default {
  data() {
    return {
      checkList: ["Camote"],
      icon: "like.svg",
      dialog: false
    };
  },
  computed: {
    ...mapGetters({ food: "food" })
  },
  created() {
    this.getFood({ id: this.$route.params.id });
  },
  methods: {
    ...mapActions({
      getFood: "GET_FOOD",
      addFood: "ADD_FOOD"
    }),
    changeLike() {
      this.icon = "Path.svg";
    },
    add() {
      this.addFood({
        id: this.$route.params.id,
        name: this.food.name,
        value: this.food.kcal
      });
      this.$router.push({ name: "Home" });
    }
  }
};
