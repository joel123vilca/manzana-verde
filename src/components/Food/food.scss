.food {
  .food__head {
    height: 60px;
    border-bottom: 2px solid #e8e8e8;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .title {
      size: 20px;
      color: #606060;
    }
    .back__button {
      color: #6b757d;
      size: 14px;
      font-weight: 600;
    }
    .iconÂ  {
      img {
        width: 30px;
      }
    }
  }
  .food__info {
    .food__image {
      margin-top: 10px;
      img {
        border-radius: 15px;
      }
    }
    .food__data {
      margin-left: 10px;
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      .food__name {
        margin-top: 10px;
        h3 {
          font-weight: 800;
          font-size: 20px;
          margin: 0px;
        }
      }
      .food__recomend {
        margin-top: 10px;
        .button__recomend {
          color: #35b266;
          border: 1px solid rgba(53, 178, 102, 0.1);
          background-color: rgba(53, 178, 102, 0.1);
          border-radius: 5px;
          width: 100px;
          height: 25px;
          margin-right: 10px;
        }
      }
      .food__description {
        margin-top: 10px;
        text-align: left;
        span {
          font-weight: bold;
        }
      }
    }

    .food__values {
      margin-left: 10px;
      margin-top: 10px;
      height: 60px;
      border: 2px solid #e8e8e8;
      border-radius: 10px;
      display: grid;
      grid-template-columns: 25% 25% 25% 25%;
      div:first-child {
        border-left: 0px solid #e8e8e8;
      }
      div {
        border-left: 2px solid #e8e8e8;
        margin-top: 10px;
        height: 40px;
        display: flex;
        flex-direction: column;
        span {
          color: #a9a9a9;
          font-size: 12px;
        }
        h3 {
          margin-top: 0px;
          color: #35b266;
        }
      }
    }
    .food__detail {
      margin-top: 10px;
      display: grid;
      grid-template-columns: 33% 33% 33%;
      .item {
        margin-top: 5px;
        display: flex;
        justify-content: center;
        label {
          color: #a9a9a9;
          font-size: 12px;
          padding-left: 5px;
        }
      }
    }
  }
  .food__additional {
    margin-top: 20px;
    margin-left: 10px;
    .additional {
      margin-top: 10px;
      border: 2px solid #e8e8e8;
      border-radius: 10px;
      height: 90px;
      display: flex;
      align-items: center;
      img {
        margin-left: 10px;
        width: 90px;
        height: 70px;
        object-fit: cover;
        border-radius: 5px;
      }
      .additional__description {
        margin-left: 20px;
        margin-top: 10px;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        span {
          color: #211e1f;
          font-size: 14px;
          font-weight: 700;
        }
        button {
          color: #35b266;
          border: 1px solid rgba(53, 178, 102, 0.1);
          background-color: rgba(53, 178, 102, 0.1);
          border-radius: 5px;
          width: 95px;
          height: 25px;
          margin-top: 5px;
          margin-bottom: 5px;
        }
        p {
          margin-top: 0px;
          color: #a9a9a9;
          font-size: 12px;
        }
      }
      .additional__action {
        margin-left: 20px;
        .add {
          width: 100px;
          height: 50px;
          font-size: 14px;
          font-weight: 700;
          border: 1px solid #ffd538;
          background: #ffd538;
          color: #606060;
          border-radius: 10px;
        }
      }
    }
  }
  .food__select {
    margin-top: 10px;
    width: 100%;
    border-bottom: 2px solid #e8e8e8;
    .checks {
      padding-left: 20px;
      display: flex;
      align-items: flex-start;
      flex-direction: column;
      .el-checkbox {
        color: #606060;
      }
      .el-checkbox__input.is-checked .el-checkbox__inner,
      .el-checkbox__input.is-indeterminate .el-checkbox__inner {
        background-color: #35b266;
        border: #35b266;
      }
      .el-checkbox__input.is-checked + .el-checkbox__label {
        color: #606060;
      }
    }
  }
  .el-checkbox-2 {
    display: flex;
    align-items: flex-start;
    padding-left: 20px;
    padding-top: 10px;
    color: #606060;
  }
  .el-checkbox__input.is-checked .el-checkbox__inner,
  .el-checkbox__input.is-indeterminate .el-checkbox__inner {
    background-color: #35b266;
    border: #35b266;
  }
  .el-checkbox__input.is-checked + .el-checkbox__label {
    color: #606060;
  }
  .food__action {
    margin-top: 10px;
    .add__order {
      width: 300px;
      height: 50px;
      font-size: 15px;
      border: 1px solid #ffd538;
      background: #ffd538;
      color: #606060;
      font-weight: 700;
      border-radius: 10px;
    }
  }
  .dialog__image {
    width: 100%;
  }
}
