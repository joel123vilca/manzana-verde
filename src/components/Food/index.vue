<template>
  <div class="food">
    <div class="food__head">
      <a @click="$router.go(-1)" class="back__button">Atras</a>
      <h2 class="title">Detalle</h2>
      <div class="icon" @click="changeLike"><img :src="require(`@/assets/${icon}`)" /></div>
    </div>
    <div class="food__info">
      <div class="food__image" @click="openDialog">
        <img :src="require('@/assets/food.png')" alt="food image" />
      </div>
      <div class="food__data">
        <div class="food__name">
          <h3>{{ food.name }}</h3>
        </div>
        <div class="food__recomend">
          <button class="button__recomend">Recomendado</button>
        </div>
        <p class="food__description">
          <span>Ingredientes:</span>
          {{ food.ingredients }}
        </p>
      </div>
      <div class="food__values">
        <div>
          <span>Kcal</span>
          <h3>{{ food.kcal }}</h3>
        </div>
        <div>
          <span>Grasa</span>
          <h3>{{ food.fat }}g</h3>
        </div>
        <div>
          <span>Carbo</span>
          <h3>{{ food.carbo }}g</h3>
        </div>
        <div>
          <span>Prote</span>
          <h3>{{ food.prote }}g</h3>
        </div>
      </div>
      <div class="food__detail">
        <div v-for="(item, index) in food.detail" :key="index" class="item">
          <img :src="require(`@/assets/${item.icon}`)" alt="item icon" />
          <label>{{ item.name }}</label>
        </div>
      </div>
    </div>
    <div class="food__additional">
      <div class="additional">
        <img :src="require('@/assets/food.png')" alt="lunch image" />
        <div class="additional__description">
          <span>Ensalada sasa</span>
          <button>Recomendado</button>
          <p>200 kcal</p>
        </div>
        <div class="additional__action">
          <button class="add">Agregar</button>
        </div>
      </div>
    </div>
    <div class="food__select">
      <el-checkbox-group v-model="checkList" class="checks">
        <el-checkbox
          v-for="(offer, index) in food.offers"
          :key="index"
          :label="offer"
          class="el-checkbox"
        ></el-checkbox>
      </el-checkbox-group>
    </div>
    <el-checkbox label="Incluir picante" class="el-checkbox-2"></el-checkbox>
    <div class="food__action">
      <button class="add__order" @click="add">Agregar al pedido</button>
    </div>
    <el-dialog :visible.sync="dialog" width="90%" center>
      <img class="dialog__image" :src="require('@/assets/food.png')" alt="food image" />
    </el-dialog>
  </div>
</template>
<script src="./food.js"></script>
<style src="./food.scss" lang="scss"></style>
