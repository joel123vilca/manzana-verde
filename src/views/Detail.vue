<template>
  <div>
    <Food />
  </div>
</template>
<script>
import Food from "@/components/Food";

export default {
  components: {
    Food
  }
};
</script>
