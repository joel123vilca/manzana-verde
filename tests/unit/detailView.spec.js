import { shallowMount, createLocalVue } from "@vue/test-utils";
import Detail from "@/views/Detail";
import Food from "@/components/Food";
import Vuex from "vuex";
import initialState from "@/store/state";
const localVue = createLocalVue();
localVue.use(Vuex);

describe("detail view", () => {
  let store, getters;
  let commit;
  let state;
  beforeEach(() => {
    store = new Vuex.Store({
      getters
    });
    commit = jest.fn();
    state = { ...initialState };
  });
  const build = () => {
    const wrapper = shallowMount(Detail, {
      store,
      localVue
    });

    return {
      wrapper,
      Food: () => wrapper.find(Food)
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the right", () => {
    const { Food } = build();
    expect(Food().exists()).toBe(true);
  });

  it("vuex in food", () => {
    const { Food } = build();
    expect(Food().vm.food).toBe(store.getters.food);
  });
});
