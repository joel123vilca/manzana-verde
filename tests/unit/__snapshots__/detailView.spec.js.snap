// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail view renders the component 1`] = `
<div>
  <food-stub></food-stub>
</div>
`;
